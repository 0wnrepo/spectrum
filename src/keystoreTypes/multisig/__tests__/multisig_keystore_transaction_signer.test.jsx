jest.mock('react-redux', () => ({
  connect: () => component => component,
}));

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MultisigKeystoreTransactionSigner from '../multisig_keystore_transaction_signer';

const web3Redux = {
  web3: () => ({
    isConnected: () => true,
    eth: {
      contract: () => ({
        at: () => ({}),
      }),
    },
  }),
};

const network = {
  id: 1,
};

describe('<MultisigKeystoreTransactionSigner />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <MultisigKeystoreTransactionSigner
        web3Redux={web3Redux}
        network={network}
        txData={{}}
        addresses={[]}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
