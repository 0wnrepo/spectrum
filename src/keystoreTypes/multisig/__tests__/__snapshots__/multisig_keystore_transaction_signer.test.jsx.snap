// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultisigKeystoreTransactionSigner /> renders correctly 1`] = `
<MultisigKeystoreTransactionProxy
  addresses={Array []}
  contract={Object {}}
  network={
    Object {
      "id": 1,
    }
  }
  txData={
    Object {
      "gas": 4000000,
    }
  }
  web3={
    Object {
      "eth": Object {
        "contract": [Function],
      },
      "isConnected": [Function],
    }
  }
  web3Redux={
    Object {
      "web3": [Function],
    }
  }
/>
`;
